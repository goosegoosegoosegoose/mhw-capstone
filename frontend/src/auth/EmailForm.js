import React, { useContext, useEffect, useState } from "react";
import UserContext from "./userContext";
import { Form, Button } from "react-bootstrap"


const EmailForm = ({edit}) => {
  const currentUser = useContext(UserContext);
  const [formData, setFormData] = useState({});
  const [email, setEmail] = useState(null);
  const INITIAL_VALUES = {
    email: "",
    password: ""
  }

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value
    }))
  };

  const handleSubmit = evt => {
    const e = false;
    try {
      evt.preventDefault();
      edit({...formData});
      setEmail(formData.email);
      setFormData(INITIAL_VALUES);
    } catch(err) {
      console.log(e)
      e = true;
      return;
    };
  };


  return (
    <div className="container">
    <div className="my-2 mx-2">
      <h2>Edit Email</h2>
      {email ? <h4>Current Email: {email}</h4> : <h4>Current Email: {currentUser.email}</h4>}
      <Form onSubmit={handleSubmit}>
        <Form.Group className="my-4">
          <Form.Label htmlFor="email">Enter desired email:</Form.Label>
          <Form.Control type="text" name="email" value={formData.email} onChange={handleChange} />
        </Form.Group>
        <Form.Group className="my-4">
          <Form.Label htmlFor="password">Confirm password to make changes:</Form.Label>
          <Form.Control type="password" name="password" value={formData.password} onChange={handleChange} />
        </Form.Group>
        <Button type="submit" variant="primary">Save New Email</Button>
      </Form>
    </div>
    </div>
  )
}

export default EmailForm;