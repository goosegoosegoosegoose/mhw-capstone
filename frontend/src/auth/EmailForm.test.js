import { render } from '@testing-library/react';
import EmailForm from "./EmailForm";

test('smoke test', () => {
  render(      
    <EmailForm />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<EmailForm />);
  expect(asFragment()).toMatchSnapshot();
});