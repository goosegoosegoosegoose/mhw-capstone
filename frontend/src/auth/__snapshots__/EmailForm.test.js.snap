// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="my-2 mx-2"
    >
      <h2>
        Edit Email
      </h2>
      <h4>
        Current Email: 
      </h4>
      <form
        class=""
      >
        <div
          class="my-4"
        >
          <label
            class="form-label"
            for="email"
          >
            Enter desired email:
          </label>
          <input
            class="form-control"
            name="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-4"
        >
          <label
            class="form-label"
            for="password"
          >
            Confirm password to make changes:
          </label>
          <input
            autocomplete="on"
            class="form-control"
            name="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Save New Email
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
